From f9bbfd4bd71d7c1b65e338b89f558a463dcac286 Mon Sep 17 00:00:00 2001
From: Florian Eckert <fe@dev.tdt.de>
Date: Thu, 20 May 2021 13:35:52 +0200
Subject: [PATCH 3/3] xr_usb_serial_common: fix compiler error for undefined
 value

The used precompiler defintion ASYNCB_INITIALIZED is not useable for new
kernel versions. This fix this use the new API to query this state.

Signed-off-by: Florian Eckert <fe@dev.tdt.de>
---
 xr_usb_serial_common-1a/xr_usb_serial_common.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xr_usb_serial_common-1a/xr_usb_serial_common.c b/xr_usb_serial_common-1a/xr_usb_serial_common.c
index 01424b6..d0c2513 100644
--- a/xr_usb_serial_common-1a/xr_usb_serial_common.c
+++ b/xr_usb_serial_common-1a/xr_usb_serial_common.c
@@ -1556,7 +1556,7 @@ static int xr_usb_serial_suspend(struct usb_interface *intf, pm_message_t messag
 	if (cnt)
 		return 0;
 
-	if (test_bit(ASYNCB_INITIALIZED, &xr_usb_serial->port.flags))
+	if (tty_port_initialized(&xr_usb_serial->port))
 		stop_data_traffic(xr_usb_serial);
 
 	return 0;
@@ -1577,7 +1577,7 @@ static int xr_usb_serial_resume(struct usb_interface *intf)
 	if (cnt)
 		return 0;
 
-	if (test_bit(ASYNCB_INITIALIZED, &xr_usb_serial->port.flags)) {
+	if (tty_port_initialized(&xr_usb_serial->port)) {
 		rv = usb_submit_urb(xr_usb_serial->ctrlurb, GFP_NOIO);
 
 		spin_lock_irq(&xr_usb_serial->write_lock);
@@ -1607,7 +1607,7 @@ err_out:
 static int xr_usb_serial_reset_resume(struct usb_interface *intf)
 {
 	struct xr_usb_serial *xr_usb_serial = usb_get_intfdata(intf);
-	if (test_bit(ASYNCB_INITIALIZED, &xr_usb_serial->port.flags)){
+	if (tty_port_initialized(&xr_usb_serial->port)) {
 #if LINUX_VERSION_CODE > KERNEL_VERSION(3, 9, 0)	
 	tty_port_tty_hangup(&xr_usb_serial->port, false);
 #else
-- 
2.20.1

